### PR TITLE
Many fixes, improvements and refactorings

### PR DESCRIPTION
**Pull request by dictoon**

### CODE REVIEW

File: src/appleseed/foundation/image/conversion.cpp [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+namespace foundation
+{
+
+bool is_linear_image_file_format(const bf::path& path)
+{
+    const string extension = lower_case(path.extension().string());
+
+    return
+        extension == ".exr"  ||
+        extension == ".tiff" ||
```
Comment (est77) [inline src/appleseed/foundation/image/conversion.cpp]: We cannot assume that TIFFs contain floating point pixels. Most do not.
Checking in Frame, we are saving linear 16 bits TIFFs, which is wrong. 
Images with integer pixels should be sRGB (I opened an issue to track that).

Comment (dictoon) [reply src/appleseed/foundation/image/conversion.cpp]: Ok.

**Comment by dictoon** [comment]
Thanks for the review.

File: src/appleseed/foundation/image/conversion.cpp (hunk 1 above)
Comment (luisbarrancos) [reply src/appleseed/foundation/image/conversion.cpp]: I'm going to be pedantic here, but you **do** have enough precision with uint16 TIFFs to have a smooth tonal progression without banding ([Barten threshold](https://www.mysterybox.us/blog/2016/10/19/hdr-video-part-3-hdr-video-terms-explained)), though using a perceptual encoding would make better use of the format.

**Pull request merged**